src/Microsoft.Framework.Runtime/NuGet/Utility/VersionUtility.cs

```diff
@@ -80,6 +88,9 @@ public static class VersionUtility
         // These aliases allow us to accept 'wp', 'wp70', 'wp71', 'windows', 'windows8' as valid target farmework folders.
         private static readonly Dictionary<FrameworkName, FrameworkName> _frameworkNameAlias = new Dictionary<FrameworkName, FrameworkName>(FrameworkNameEqualityComparer.Default)
         {
+            { new FrameworkName("Asp.Net, Version=v5.0"), new FrameworkName("DNX, Version=v4.5.1") },
```

Comment:
It's the same feature that was used to rename "sl4-wp71" to "wp71". We're trying to rename a framework here, so it's the same concept.

Basically all this affects is that BOTH frameworks that come in to IsCompatible go through this dictionary before comparison. Thus, aspnet50 is compared as identical to dnx451, etc.